### DISCUSSION TITLE
How producer to know whether the status Kafka cluster

### DISCUSSION THREAD

**Discussion by barryzhounb** [strimzi/strimzi-kafka-operator#4656] [General]
Currently we are using Strimizi as out Kafka cluster, I have some question about producer.

when we initialize a producer as following, after that, then Kafka producer is initialized, but at this time, suppose Kafka cluster is not available, we could not know the status Kafka cluster from producer.

```
producer = new KafkaProducer<>(properties);
```
Once we call producer.send(new ProducerRecord()), then we get exception such as TimeoutExeption.

I am wondering what is proposed pattern to implement producer in order to know the status Kafka cluster. In the above scenario, we don't know the status Kafka cluster until we hit the error when we send message.

Is there a way to get the status Kafka cluster  when we initialize a producer or before we call producer.send() to send message?

Is there any  proposed pattern?

**Comment by barryzhounb** [comment]
Hi @scholzj , any ideas or suggestions for this question?

**Reply by scholzj** [reply]
No, not really. Sorry. I guess you can use something from the AdminClient.

**Reply by barryzhounb** [reply]
thanks
